dandi/upload.py

```diff
@@ -61,6 +61,21 @@ def upload(
             " paths.  Use 'dandi download' or 'organize' first."
         )
 
+    # pre-validate BIDS datasets before going for individual
+    # files etc
+    bids_datasets = _bids_discover_and_validate(dandiset_.path, paths, validation)
+
+    if bids_datasets:
+        _bids_datasets = [str(i) for i in bids_datasets]
+        if not allow_any_path:
+            lgr.info(
+                "Setting allow_any_path to True since we detected %s under the "
```

Comment:
```suggestion
                "Enabling --allow-any-path since we detected %s under the "
```

(Because this function is most often invoked via the command line.)